app/src/main/java/com/example/foodie/common/data/mapper/Mapper.kt

```diff
@@ -5,86 +5,62 @@ import com.example.foodie.common.data.model.FAT
 import com.example.foodie.common.data.model.FoodResponce
 import com.example.foodie.common.data.model.Hit
 import com.example.foodie.common.data.model.Images
-import com.example.foodie.common.data.model.Links
 import com.example.foodie.common.data.model.PROCNT
-import com.example.foodie.common.data.model.REGULAR
 import com.example.foodie.common.data.model.Recipe
-import com.example.foodie.common.data.model.Self
 import com.example.foodie.common.data.model.TotalNutrients
-import com.example.foodie.common.data.model.CHOCDFDTO
-import com.example.foodie.common.data.model.FATDTO
-import com.example.foodie.common.data.model.FoodResponseDTO
-import com.example.foodie.common.data.model.HitDTO
-import com.example.foodie.common.data.model.LinksDTO
-import com.example.foodie.common.data.model.ImagesDTO
+import com.example.foodie.common.data.modelDTO.CHOCDFDTO
+import com.example.foodie.common.data.modelDTO.FATDTO
+import com.example.foodie.common.data.modelDTO.FoodResponseDTO
+import com.example.foodie.common.data.modelDTO.HitDTO
+import com.example.foodie.common.data.modelDTO.ImagesDTO
 import com.example.foodie.common.data.model.LARGE
-import com.example.foodie.common.data.model.LARGEDTO
-import com.example.foodie.common.data.model.PROCNTDTO
-import com.example.foodie.common.data.model.REGULARDTO
-import com.example.foodie.common.data.model.RecipeDTO
-import com.example.foodie.common.data.model.SelfDTO
-import com.example.foodie.common.data.model.TotalNutrientsDTO
+import com.example.foodie.common.data.modelDTO.LARGEDTO
+import com.example.foodie.common.data.modelDTO.PROCNTDTO
+import com.example.foodie.common.data.modelDTO.RecipeDTO
+import com.example.foodie.common.data.modelDTO.TotalNutrientsDTO
 
 
 fun FoodResponseDTO.toFoodResponce(): FoodResponce {
-    return FoodResponce(_links = this._links?.toLinks() ?: Links(
-        self = Self(
-            "", ""
-        )
-    ),
+    return FoodResponce(
         count = this.count ?: 0,
         from = this.from ?: 0,
         hits = this.hits?.map { it.toHit() } ?: emptyList(),
         to = this.to ?: 0)
 }
 
-fun LinksDTO.toLinks(): Links {
-    return Links(
-        self = this.self?.toSelf() ?: Self(href = "", title = "")
-    )
-}
-
-fun SelfDTO.toSelf(): Self {
-    return Self(
-        href = this.href ?: "", title = this.title ?: ""
-    )
-}
 
 
 fun HitDTO.toHit(): Hit {
     return Hit(
-        recipe = this.recipe?.toRecipe() ?: Recipe(
-            0.0, "", Images(
-                LARGE(0, "", 0),
-                REGULAR(0, "", 0),
-            ),
-
-            emptyList(), "", emptyList(), TotalNutrients(
-                CHOCDF("", 0.0, ""),
-                FAT("", 0.0, ""),
-                PROCNT("", 0.0, ""),
-            ), 0.0, 0.0, "", ""
-        )
+        recipe = this.recipe.toRecipe()
     )
 }
 
 
-fun RecipeDTO.toRecipe(): Recipe {
+fun RecipeDTO?.toRecipe(): Recipe {
+    if (this == null) {
```

Comment:
rework creation depending of null parameters and not of null class
example: this?.X ?: Y
it will reduce the amount of boilerplate code